cairo_zero/kakarot/precompiles/kakarot_precompiles.cairo

```diff
@@ -139,3 +153,193 @@ namespace KakarotPrecompiles {
+    // @ param skip_returndata Whether to skip returning the returndata of the starknet call
+    func execute_cairo_call{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr,
+        bitwise_ptr: BitwiseBuiltin*,
+    }(
+        caller_address: felt,
+        evm_selector: felt,
+        to_starknet_address: felt,
+        starknet_selector: felt,
+        data_len: felt,
+        data: felt*,
+        skip_returndata: felt,
+    ) -> (output_len: felt, output: felt*, gas_used: felt, reverted: felt) {
+        alloc_locals;
+        let is_evm_selector_valid_ = is_evm_selector_valid(evm_selector);
+        if (is_evm_selector_valid_ == FALSE) {
+            let (revert_reason_len, revert_reason) = Errors.invalidEvmSelector();
+            return (
+                revert_reason_len, revert_reason, CAIRO_PRECOMPILE_GAS, Errors.EXCEPTIONAL_HALT
+            );
+        }
+
+        let caller_starknet_address = Account.get_registered_starknet_address(caller_address);
+        let is_not_deployed = Helpers.is_zero(caller_starknet_address);
+
+        if (is_not_deployed != FALSE) {
+            // Deploy account -
+            // order of returned values in memory matches the explicit ones in the other branch
+            Starknet.deploy(caller_address);
+        } else {
+            tempvar syscall_ptr = syscall_ptr;
+            tempvar pedersen_ptr = pedersen_ptr;
+            tempvar range_check_ptr = range_check_ptr;
+            tempvar caller_starknet_address = caller_starknet_address;
+        }
+        let syscall_ptr = cast([ap - 4], felt*);
+        let pedersen_ptr = cast([ap - 3], HashBuiltin*);
+        let range_check_ptr = [ap - 2];
+        let caller_starknet_address = [ap - 1];
+
+        let (retdata_len, retdata, success) = IAccount.execute_starknet_call(
+            caller_starknet_address, to_starknet_address, starknet_selector, data_len, data
+        );
+        if (success == FALSE) {
+            // skip formatting to bytes32 array and return revert reason directly
+            return (retdata_len, retdata, CAIRO_PRECOMPILE_GAS, Errors.EXCEPTIONAL_HALT);
+        }
+
+        let (output) = alloc();
+        if (skip_returndata != FALSE) {
+            return (0, output, CAIRO_PRECOMPILE_GAS, FALSE);
+        }
+
+        let output_len = retdata_len * 32;
+        Helpers.felt_array_to_bytes32_array(retdata_len, retdata, output);
+        return (output_len, output, CAIRO_PRECOMPILE_GAS, FALSE);
+    }
+
+    // @notice Checks if the selector is a valid EVM selector for a Cairo call.
+    // @dev Currently, two selectors are supported:
+    // call_contract(uint256,uint256,uint256[]) - for "regular" CallCairo Precompile
+    // call_contract(uint256,uint256,uint256,uint256[]) - for MulticallCairo Precompile
+    // @param selector The selector to check.
+    // @returns TRUE if the selector is valid, FALSE otherwise.
+    func is_evm_selector_valid(selector: felt) -> felt {
```

Comment:
`evm_selector` is a legacy from the time when we did both `contract` and `library` calls at the same precompile address, it's no longer required